packages/discovery-provider/plugins/pedalboard/apps/mri/src/clm.ts

```diff
@@ -0,0 +1,70 @@
+import { Knex } from "knex";
+import { logger as plogger } from "./logger";
+import { toCsvString } from "./csv";
+import { ClmS3Config, publishToS3 } from "./s3";
+
+export type ClientLabelMetadata = {
+    UniqueTrackIdentifier: number;
+    TrackTitle: string;
+    Artist: string;
+    AlbumTitle: string;
+    AlbumId: number;
+    ReleaseLabel: string;
+    ISRC: string;
+    UPC: string | null;
+    Composer: string;
+    Duration: number;
+    ResourceType: string;
+}
+
+// gathers data from a 24 hour period between "YYYY-MM-DDT00:00:00.000Z" to "YYYY-MM-DDT23:59:59.999Z"
+// formats it into csv format compatible with the mri spec
+// publishes it to all the provided s3 configs
+export const clm = async (db: Knex, s3s: ClmS3Config[], date: Date): Promise<void> => {
+    const logger = plogger.child({ date: date.toISOString() })
+    logger.info("beginning report processing")
+
+    // Gather data from "YYYY-MM-DDT00:00:00.000Z" to "YYYY-MM-DDT23:59:59.999Z"
+    const start = new Date(date);
+    start.setHours(0, 0, 0, 0)
+    const end = new Date(date);
+    end.setHours(23, 59, 59, 999)
+
+    logger.info({ start: start.toISOString(), end: end.toISOString() }, "time range")
+
+    const clmRows: ClientLabelMetadata[] = await db('tracks')
+        .join(
+            'users',
+            'tracks.owner_id', '=', 'users.user_id'
+        )
+        .leftJoin('playlist_tracks', 'tracks.track_id', '=', 'playlist_tracks.track_id')
+        .leftJoin('playlists', function () {
+            this.on('playlist_tracks.playlist_id', '=', 'playlists.playlist_id')
+                .andOn('playlists.is_album', '=', db.raw('true'))
+        })
+        .distinctOn('tracks.track_id')
+        .select(
+            `tracks.track_id as UniqueTrackIdentifier`,
+            `tracks.title as TrackTitle`,
+            `users.name as Artist`,
+            `playlists.playlist_name as AlbumTitle`,
+            `playlists.playlist_id as AlbumId`,
+            db.raw(`'' as "ReleaseLabel"`),
+            `tracks.isrc as ISRC`,
+            `playlists.upc as UPC`,
+            db.raw(`'' as "Composer"`),
+            `tracks.duration as Duration`,
+            db.raw(`'Audio' as "ResourceType"`)
+        )
+        .whereNotNull('playlists.playlist_id')
+        .whereNotNull('isrc')
+        .andWhereNot('isrc', '=', '')
+        .where('tracks.created_at', '>=', start)
+        .where('tracks.created_at', '<', end)
```

Comment:
since this is `<` there is a possibility (though tiny tiny) to miss a track.

  why not define

```
end = currentDate.setDate(currentDate.getDate() + 1);
```

and keep this as `<`, that way we will get everything